src/application/app.rs

```diff
@@ -4,13 +4,21 @@ use crate::{application::State, backend::db};
 pub enum InputMode {
     Command,
     AddTask,
+    Prompt,
+}
+
+#[derive(PartialEq)]
+pub enum Actions {
+    AddTask,
+    ClearAllTasks,
 }
 
 pub struct App {
     pub is_running: bool,
     pub state: State,
     pub text_input: String,
     pub input_mode: InputMode,
+    pub previous_action: Actions,
```

Comment:
I suggest changing previous action to be Option<Actions> instead, this way it will be easier to initialize (None) and will have meaning in context where user didn't do any action yet

Reply:
done